substrate/frame/election-provider-multi-block/src/signed/tests.rs

```diff
@@ -224,16 +231,364 @@ mod calls {
+			);
+
+			assert_err!(
+				SignedPallet::bail(RuntimeOrigin::signed(account_id)),
+				NotAcceptingSubmissions::<Runtime>,
+			);
+		})
+	}
+
+	#[test]
+	fn register_and_submit_page_and_bail_prohibitted_in_phase_unsigned() {
+		ExtBuilder::default().build_and_execute(|| {
+			let some_bn = 0;
+
+			let account_id = 99;
+			roll_to_phase(Phase::Unsigned(some_bn));
+
+			assert_err!(
+				SignedPallet::register(RuntimeOrigin::signed(account_id), Default::default()),
+				NotAcceptingSubmissions::<Runtime>,
+			);
+
+			assert_err!(
+				SignedPallet::submit_page(
+					RuntimeOrigin::signed(account_id),
+					0,
+					Some(Default::default())
+				),
+				NotAcceptingSubmissions::<Runtime>,
+			);
+
+			assert_err!(
+				SignedPallet::bail(RuntimeOrigin::signed(account_id)),
+				NotAcceptingSubmissions::<Runtime>,
+			);
+		})
+	}
+
+	#[test]
+	fn register_and_submit_page_and_bail_prohibitted_in_phase_snapshot() {
+		ExtBuilder::default().build_and_execute(|| {
+			let some_page_index = 0;
+			let account_id = 99;
+			roll_to_phase(Phase::Snapshot(some_page_index));
+
+			assert_err!(
+				SignedPallet::register(RuntimeOrigin::signed(account_id), Default::default()),
+				NotAcceptingSubmissions::<Runtime>,
+			);
+
+			assert_err!(
+				SignedPallet::submit_page(
+					RuntimeOrigin::signed(account_id),
+					0,
+					Some(Default::default())
+				),
+				NotAcceptingSubmissions::<Runtime>,
+			);
+
+			assert_err!(
+				SignedPallet::bail(RuntimeOrigin::signed(account_id)),
+				NotAcceptingSubmissions::<Runtime>,
+			);
+		})
+	}
+
+	#[test]
+	fn register_and_submit_page_and_bail_prohibitted_in_phase_export() {
+		ExtBuilder::default().build_and_execute(|| {
+			let some_bn = 0;
+			let account_id = 99;
+			roll_to_phase(Phase::Export(some_bn));
+
+			assert_err!(
+				SignedPallet::register(RuntimeOrigin::signed(account_id), Default::default()),
+				NotAcceptingSubmissions::<Runtime>,
+			);
+
+			assert_err!(
+				SignedPallet::submit_page(
+					RuntimeOrigin::signed(account_id),
+					0,
+					Some(Default::default())
+				),
+				NotAcceptingSubmissions::<Runtime>,
+			);
+
+			assert_err!(
+				SignedPallet::bail(RuntimeOrigin::signed(account_id)),
+				NotAcceptingSubmissions::<Runtime>,
+			);
+		})
+	}
+
+	#[test]
+	fn register_and_submit_page_and_bail_prohibitted_in_phase_emergency() {
+		ExtBuilder::default().build_and_execute(|| {
+			let account_id = 99;
+			compute_snapshot_checked();
+			roll_to_phase(Phase::Emergency);
+
+			assert_err!(
+				SignedPallet::register(RuntimeOrigin::signed(account_id), Default::default()),
+				NotAcceptingSubmissions::<Runtime>,
+			);
+
+			assert_err!(
+				SignedPallet::submit_page(
+					RuntimeOrigin::signed(account_id),
+					0,
+					Some(Default::default())
+				),
+				NotAcceptingSubmissions::<Runtime>,
+			);
+
+			assert_err!(
+				SignedPallet::bail(RuntimeOrigin::signed(account_id)),
+				NotAcceptingSubmissions::<Runtime>,
+			);
+		})
+	}
+
+	#[test]
```

Comment:
I was rather testing what the test name says, so: given there are existing submissions, if someone bails from a submission that they didn't do, then the balances of all the participants, the one that had a submission and especially the one that bailed but didn't have one will remain unchanged.
Maybe it's too much of a defensive approach but I felt like it's worth having coverage there.
So that's my reasoning but if you don't like it here, then I'll remove it, sure.

Reply:
> so: given there are existing submissions, if someone bails from a submission that they didn't do, then the balances of all the participants, the one that had a submission and especially the one that bailed but didn't have one will remain unchanged.

This can be tested using the `assert_noop!` macro. Basically, when `assert_noop!(call(), Error)`, the macro checks a couple things:
1. The `call()` will return the expected `Error
2. The Ext storage was not changed after calling `call()`. this means that the balances, and other states remain the same.

I'd just remove the whole set up of the registration etc and call an `assert_noop!(bail_for(1), Err)`. This tests that the bailing does not work in this case and the storage root remained the same, so no state has changed (+ simplify the tests considerably).